internal/ingress/controller/controller.go

```diff
@@ -1115,8 +1123,17 @@ func (n *NGINXController) createServers(data []*ingress.Ingress,
 	return servers
 }
 
+// OK to merge canary ingresses iff there exists one or more ingresses to potentially merge into
+func safeToMergeCanaryIngresses(ingresses []*ingress.Ingress, canaryIngresses []*ingress.Ingress) bool {
+	return len(ingresses)-len(canaryIngresses) > 0
```

Comment:
IMO this is a bit confusing because main safety is being done in `mergeAlternativeBackends` function. I'd suggest to call this something like `nonCanaryIngressExists`